src/api/spec/models/comment_spec.rb

```diff
@@ -0,0 +1,88 @@
+require "rails_helper"
+
+RSpec.shared_examples "a comment" do
+  let(:comment_factory) { described_class.name.underscore }
+  let(:comment) { create(comment_factory) }
+
+  describe "A comment" do
+    it { expect(build(comment_factory)).to be_valid }
+  end
+
+  describe "Comment associations" do
+    it { is_expected.to belong_to(:bs_request).inverse_of(:comments) }
+    it { is_expected.to belong_to(:project).inverse_of(:comments) }
+    it { is_expected.to belong_to(:package).inverse_of(:comments) }
+    it { is_expected.to belong_to(:user).inverse_of(:comments) }
+
+    it { is_expected.to have_many(:children).dependent(:destroy).class_name('Comment').with_foreign_key('parent_id') }
+  end
+
+  describe "Comment validations" do
+    it { is_expected.to validate_presence_of(:body) }
+    it { is_expected.to validate_presence_of(:type) }
+    it { is_expected.to validate_presence_of(:user) }
+  end
+
+  describe "to_xml" do
+    context "without parent" do
+      let(:builder) { builder = Nokogiri::XML::Builder.new }
+
+      before do
+        comment.to_xml(builder)
+      end
+
+      context "returns xml" do
+        context "comment element" do
+          let(:comment_element) { builder.doc.css('comment') }
+
+          it "comment attributes and content" do
+            expect(comment_element.attribute('id').value).to eq(comment.id.to_s)
+            expect(comment_element.attribute('when').value).to match(/^\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d UTC$/)
+            expect(comment_element.attribute('who').value).to match(/^user_\d+$/)
+
+            expect(comment_element.text).to match(/^#<Nokogiri::XML::Builder::NodeBuilder:0x\h+>$/)
+          end
+        end
+      end
+    end
+
+    context "with parent" do
```

Comment:
This tests is exactly the same that the one above but with one more attribute. Why have you left it empty? I would prefer to have the `with parent` case as it is "more general".